lib/sheet.js

```diff
@@ -0,0 +1,76 @@
+import Spreadsheet from 'edit-google-spreadsheet';
+import googleAuth from './google-auth';
+import util from './util';
+
+let SHEET_CONFIG;
+util.readJsonFromFile('./google-spreadsheet-config.json', data => {
+    SHEET_CONFIG = data;
+});
+
+class Sheet {
+    constructor() {
+        this.dataCache = {};
+        return this;
+    }
+
+    fetch(searchString, processingCallback, afterRequestCallback){
+        if (!this.dataCache.extensions) {
+            this.updateCache(searchString, processingCallback, afterRequestCallback);
+        }else{
+            processingCallback(searchString, this.dataCache, afterRequestCallback);
+        }
+    }
+
+    updateCache(searchString, processingCallback, afterRequestCallback){
+        googleAuth.authenticate(tokens => {
+            this.requestSpreadsheet(SHEET_CONFIG.extensionsWorksheetId, SHEET_CONFIG.spreadsheetId, tokens, (extensions) => {
+                this.requestSpreadsheet(SHEET_CONFIG.aliasesWorksheetId, SHEET_CONFIG.spreadsheetId, tokens, (aliases) => {
+                    this.dataCache = this.buildData(extensions, aliases);
+                    if (processingCallback){
+                        processingCallback(searchString, this.dataCache, afterRequestCallback);
+                    }
+                });
+            });
+        });
+    }
+
+    requestSpreadsheet(worksheetId, spreadsheetId, tokens, callback){
```

Comment:
Since `spreadsheetId` and `tokens` never change, we could read them directly from `SHEET_CONFIG` and reduce the amount of parameters in this method. What do you think?